decompiler/pipeline/dataflowanalysis/dead_path_elimination.py

```diff
@@ -80,6 +96,32 @@ def _is_invalid_edge(self, edge: BasicBlockEdge, condition: BoolRef) -> bool:
             condition = Not(condition)
         return self._logic_converter.is_not_satisfiable(condition, timeout=self._timeout)
 
+    def _is_invalid_edge_delogic(self, edge: BasicBlockEdge, condition: WorldObject) -> bool:
+        """
+        Check whether the condition of the given branch is satisfyable or not.
+
+        Returns a tuple of bools indicating the viability of the true and false branches.
+        """
+        if isinstance(edge, FalseCase):
+            condition = self._logic_converter.negate(condition)
+        result = self._logic_converter.make_variable(condition)
+        result.simplify()
+        # simplify to fixpoint.
+        while True:
+            if hasattr(result, "operands"):
+                result_copy = self._logic_converter.make_variable(result.copy_tree())
+                result.simplify()
+                if self._logic_converter.compare(result_copy, result):
+                    break
+            else:
+                break
+        if result is None:
+            return True
+        if hasattr(result, "operands"):
+            if len(result.operands) == 0:
+                return True
```

Comment:
I am a bit confused, how can the result be None, and what is the case for the second check? Why not ask whether it is an operation. Each operation should have operands or?

Reply:
You're right, I forgot that I removed some changes to ```simplify```. :sweat: 
For the operands, I was thinking of a way to not import structures from the simplifier into this file. However, you're right that if we implement the ```check``` and ```is_satisfiable```, then I don't need to worry about that. Let me make the corresponding changes.